src/Google.Pubsub.V1/PublisherClient.cs

```diff
@@ -423,6 +492,30 @@ public virtual Publisher.IPublisherClient GrpcClient
 
     }
 
+    // Something like this will be in GAX
+    public class ApiCall<TRequest, TResponse>
+        where TRequest : class, IMessage<TRequest>
+        where TResponse : class, IMessage<TResponse>
+    {
+        // May or may not keep these delegates. Using Func<>s instead is a little ugly.
```

Comment:
I'm about to do the GAX change for all of this.
I'll see how often these Funcs would be used, and decide if it works better with or without the delegates.
Clearly, the functionality isn't affected by the decision, but readability is ;)